### PR TITLE
Optimize built-in generation performance

### PR DESCRIPTION
**Pull request by koxudaxi**
## Summary by CodeRabbit

- **Performance**
  - Improved formatting speed for generated code by efficiently processing simple import sections.
  - Preserved existing behavior for complex or unsupported code through automatic fallback formatting.

- **Bug Fixes**
  - Improved handling of malformed, incomplete, compound, and import-only generated code.
  - Added compatibility checks for formatting options and Python versions.

- **Tests**
  - Expanded coverage for generated-code formatting edge cases and fallback scenarios.
  - Added performance coverage for large OpenAPI schemas.
  - Verified consistent output across supported formatting configurations.

### CODE REVIEW

File: src/datamodel_code_generator/format.py [hunk 1]
```diff
@@ -711,10 +712,27 @@
...
         if Formatter.ISORT in self.formatters:
             code = self.apply_isort(code)
         if self.use_builtin_formatter:
-            code = self.apply_builtin_formatter(
+            match generated:
+                case True:
+                    formatter = _builtin_formatter_attr("_apply_builtin_generated_formatter")
+                case False:
+                    formatter = self.apply_builtin_formatter
+            code = formatter(
```
[the remark this answers is not in the document]
Comment (koxudaxi) [reply src/datamodel_code_generator/format.py:736]: Fixed in d936462a by initializing the formatter before the generated-code dispatch. The latest CodeQL analysis now passes.
